packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2190,6 +2249,20 @@ export class ContainerRuntime
 	}
 
 	public setConnectionState(connected: boolean, clientId?: string) {
+		if (connected && this.idCompressorMode === "delayed" && !this.delayedCompressorLoading) {
+			this.delayedCompressorLoading = true;
+			this.createIdCompressor()
+				.then((compressor) => {
+					this._idCompressor = compressor;
+					for (const range of this.pendingIdCompressorOps) {
+						this._idCompressor.finalizeCreationRange(range);
+					}
+					this.pendingIdCompressorOps = [];
+				})
+				.catch((error) => {
+					this.logger.sendErrorEvent({ eventName: "IdCompressorDelayedLoad" }, error);
```

Comment:
Shouldn't we just close the container here? Or I guess the idea is that if compressor isn't there, we will use regular ids and move on?

Reply:
Yes, regular IDs will be fine. There is no other usage for ID compressor, so there is no issue with correctness.